src/components/service/face_detect.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {Layout, Divider, Card, Icon, Spin, Alert, Row, Col, Button, Tag, message, Table, Collapse, Steps, Modal, Upload} from 'antd';
+
+class FaceDetectService extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.submitAction = this.submitAction.bind(this);
+    this.state = {
+        fileUploaded: false,
+        file: undefined,
+        fileReader: undefined,
+        methodName: "find_face",
+    };
+  }
+
+  isComplete() {
+    if (this.props.jobResult === undefined)
+        return false;
+    else
+    {
+        console.log(this.props.jobResult);
+        return true;
+    }
+  }
+  
+  processFile(file) {
+    let reader = new FileReader();
+
+    reader.onload = (e => {
+      this.setState({
+        fileUploaded: true,
+        file: file,
+        fileReader: reader,
+      });
+    });
+
+    reader.readAsDataURL(file);
+  }
+
+  submitAction() {
+    this.props.showModalCallback(this.props.callModal);
+    this.props.callApiCallback(this.state.methodName, 
+      {
+        image: this.state.fileReader.result.split(',')[1],
+      }
+    );
+  }
+
+  renderBoundingBox(result)
```

Comment:
Looks like this function is used in three places, four if you count the slightly modified renderLandmarks. Do you want to refactor it out to a canvas utils class? You could pass parameters for which key to look up from the result object and a lambda for what function to apply to the objects in the forEach.

Reply:
I forgot I left those in. Only face_detect.js uses renderBoundingBox so I'll remove the others.

One reason for keeping these as self-contained as possible is that we'll eventually want service developers to define the UI (or the dapp will generate the UI from the API definition).